code/modules/tgui/states/access.dm

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) 2020 Aleksej Komarov
+ * SPDX-License-Identifier: MIT
+ */
+
+/**
+ * tgui state: access_state
+ *
+ * Humans need to have access and be adjacent to use it.
+ * Silicons and other lifeforms get their default ui_state pass.
+ */
+
+GLOBAL_DATUM_INIT(access_state, /datum/ui_state/access_state, new)
+
+/datum/ui_state/access_state/can_use_topic(src_object, mob/user)
+	return user.access_can_use_topic(src_object)
+
+/mob/proc/access_can_use_topic(src_object)
+	return src.default_can_use_topic(src_object)
+
+/mob/living/access_can_use_topic(src_object)
+	. = src.human_adjacent_can_use_topic(src_object)
+
+	var/obj/O = src_object
+	if(!O?.allowed(src)) //No access? No ui!
```

Comment:
if this is not going to allow you to see but not interact then you should just add that check on ui_interact like other ui's do.

Reply:
When i just do a if(Mob.allowed(src)) check in ui_interact instead of this ui_state then that check becomes universal, meaning ghosts would not be able to view the console at all, as is the case for a couple other things around.